arch/posix/core/posix_core.c

```diff
@@ -0,0 +1,272 @@
+ *  (somebody calls posix_core_let_run() with this thread number
+ */
+static void pc_wait_until_allowed(void)
+{
+	pthread_t this_thread_nbr = pthread_self();
+
+	while (!pthread_equal(this_thread_nbr,
+		pc_currently_allowed_thread)) {
+
+		pthread_cond_wait(&pc_cond_threads,
+				  &pc_mtx_threads);
+	}
+/*
+ * Note that we go out of this while loop with the mutex locked by this
+ * particular thread. The mutex is only unlocked internally in
+ * pthread_cond_wait() [in posix_core_swap()] while waiting for cond to be
+ * signaled
+ */
+
+	if (POSIX_ARCH_DEBUG_PRINTS) {
+		ps_print_trace(
+			"Posix arch core: Thread %lu: I'm allowed to run!\n",
+			this_thread_nbr);
+	}
+}
+
+
+/**
+ *  Helper function to let the thread with id <next_allowed_thread_nbr> run
+ *  Note: posix_core_let_run() can only be called with the mutex locked
+ */
+static void pc_let_run(p_thread_id_t next_allowed_thread_nbr)
+{
+	if (POSIX_ARCH_DEBUG_PRINTS) {
+		ps_print_trace("Posix arch core: We let thread %lu run\n",
+				next_allowed_thread_nbr);
+	}
+
+	pc_currently_allowed_thread = next_allowed_thread_nbr;
+
+	if (pthread_cond_broadcast(&pc_cond_threads)) {
+/*
+ * we let all threads know one is able to run now (it may even be us again
+ * if fancied)
+ * Note that as we hold the mutex, they are going to be blocked until we reach
+ * our own wait_util_allowed() while loop
+ */
+		ps_print_error_and_exit(
+			"Posix arch core: error on pthread_cond_signal()\n");
+	}
+}
+
+/**
+ * Let the ready thread run and lock this thread until it is allowed again
+ *
+ * Note that <next_allowed_thread_nbr> is an opaque type
+ * representing a thread id/number which only needs to be understood
+ * by this file functions
+ *
+ * called from __swap() which does the picking from the kernel structures
+ */
+void pc_swap(p_thread_id_t next_allowed_thread_nbr)
+{
+	pc_let_run(next_allowed_thread_nbr);
+	pc_wait_until_allowed();
+}
+
+/**
+ * Let the ready thread (main) run and exit (init)
+ *
+ * Called from _arch_switch_to_main_thread() which does the picking from the
+ * kernel structures
+ */
+void posix_core_main_thread_start(p_thread_id_t next_allowed_thread_nbr)
+{
+	if (0) {
+		pc_swap(next_allowed_thread_nbr);
+	} else {
+		pc_let_run(next_allowed_thread_nbr);
+
+		if (pthread_mutex_unlock(&pc_mtx_threads)) {
+			ps_print_error_and_exit(
+			"Posix arch core: error on pthread_mutex_lock()\n");
+		}
+
+		pthread_exit(NULL);
+/*
+ * The only reason to have this is to be a bit more tidy and not let the init
+ * thread lingering around
+ */
+	}
+}
+
+
+/**
+ * Helper function to start a Zephyr thread as a POSIX thread:
+ *  It will block the thread until a __swap() is called for it
+ *
+ * Spawned from posix_core_new_thread() below
+ */
+static void *pc_thread_starter(void *arg)
+{
+	posix_thread_status_t *ptr = (posix_thread_status_t *) arg;
+
+	/*we detach ourselves so nobody needs to join to us*/
+	pthread_detach(pthread_self());
+
+	if (pthread_mutex_lock(&pc_mtx_threads)) {
+/*
```

Comment:
comment blocks inside functions should have the indentation aligned

Reply:
Will do